pkg/gzip/error.go

```diff
@@ -0,0 +1,52 @@
+package gzip
+
+import (
+	"fmt"
+	"io"
+)
+
+type errType int
+
+const (
+	info     errType = 0
+	skipping errType = 1
+	warning  errType = 2
+	fatal    errType = 3
+)
+
+type appError struct {
+	msg   string
+	level errType
+	path  string
+}
+
+func (e *appError) Error() string {
+	if e.path != "" {
+		return e.path + " " + e.msg
+	}
+	return e.msg
+}
+
+func ErrorHandler(err error, stdout io.Writer, stderr io.Writer) int {
+	switch err.(type) {
+	case *appError:
+		switch err.(*appError).level {
```

Comment:
Rather than switching on the level here, appError.Error should just return the right error string. Then, you need not have the ErrorHandler at all, which seems like a little too much for this

Reply:
Removed all of error.go, and simplified error handling across the app. The only thing that I didn't carry over is exit status 0 on gzip -h, due to where the help option is parsed.